src/TeleSign.Services/TeleSignService.cs

```diff
@@ -52,6 +52,7 @@ public class TeleSignService
                     IWebRequester webRequester,
                     string accountName = "default")
         {
+            ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls12;
```

Comment:
Unfortunately, my research landed us in the same place as you. A new app domain would be required to change the value of this configuration. That also leads to it's own set of challenges ensuring all types can be properly marshaled across the app domain boundary.

That being said, it would still be my preference if libraries did not make this choice for me as it has consequences on any other code running in my app domain. While I don't know the TeleSign policy specifically, but if TLS 1.2 is required, I'd prefer an approach that threw an exception if TLS 1.2 wasn't a configured value because then It's my choice to decide how to deal with it (app domain or otherwise). Moving me to TLS 1.2 inside the library can likely introduce other bugs in my code (i.e. i have specific services I need to talk to that support TLS 1.1 or SSLv3)

It's also worth mentioning that the SecurityProtocol enum is technically a bitmask of representing allowed operations. Setting it to TLS 1.2 like this PR suggests rules out the other versions (as I understand how the Microsoft libraries are meant to work).